src/normalize/meta.rs

```diff
@@ -21,26 +22,24 @@ type BuildId = Vec<u8>;
 /// let normalized = normalizer
 ///     .normalize_user_addrs_sorted([addr_in_elf_in_apk].as_slice(), Pid::Slf)
 ///     .unwrap();
-/// let (norm_addr, meta_idx) = normalized.addrs[0];
+/// let (output, meta_idx) = normalized.outputs[0];
 /// let meta = &normalized.meta[meta_idx];
-/// let apk_elf = meta.apk_elf().unwrap();
+/// let apk = meta.apk().unwrap();
 ///
-/// // Let's assume we have the ELF file lying around in a hypothetical
-/// // data/ directory.
-/// let elf_path = Path::new("data/").join(&apk_elf.elf_path);
-///
-/// let src = symbolize::Source::from(symbolize::Elf::new(elf_path));
+/// // We assume that we have the APK lying around at the same path as on the
+/// // "remote" system.
+/// let src = symbolize::Source::from(symbolize::Apk::new(&apk.path));
 /// let symbolizer = symbolize::Symbolizer::new();
-/// let syms = symbolizer.symbolize(&src, &[norm_addr]).unwrap();
+/// let sym = symbolizer
+///   .symbolize_single(&src, symbolize::Input::FileOffset(output))
+///   .unwrap()
+///   .into_sym()
+///   .unwrap();
 /// ```
 #[derive(Clone, Debug, PartialEq)]
-pub struct ApkElf {
+pub struct Apk {
     /// The canonical absolute path to the APK, including its name.
-    pub apk_path: PathBuf,
-    /// The relative path to the ELF file inside the APK.
-    pub elf_path: PathBuf,
-    /// The ELF file's build ID, if available.
-    pub elf_build_id: Option<BuildId>,
```

Comment:
this might be something that we already discussed and concluded, if that's the case, I'm sorry for forgetting. But I was under impression that we still will wanted to keep optional BuildId for APK for cases where it's fine to do a bit more work locally to get build_id (similar to have we can still get build_id locally, as an opt-in option, for just normal non-APK binaries). Just like with non-APK case, it should be some option for normalization API that would allow going through `/proc/<pid>/map_files/<start>_<end> files` in the name of better data (build ID mainly).

Reply:
I thought our conclusion was that we wanted to eliminate build ID logic for APK altogether, given that we had to introduce the APK symbolization source anyway. We could bring it back, but I am not particularly inclined to do so given that it adds a bunch of complexity and I don't think the one consumer of the APK logic has any intention (or ability) to use that :shrug:

Reply:
ok, no big deal, the asymmetry with ELF case was what triggered this question, but you are right, we can always add it later if we have use cases